planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp

```diff
@@ -1181,22 +1214,81 @@ AvoidLineArray AvoidanceModule::extractShiftLinesFromLine(ShiftLineData & shift_
     }
 
     if (!found_first_start) {
-      utils::avoidance::setStartData(
-        al, 0.0, p, i, arcs.at(i));  // start length is overwritten later.
+      setStartData(al, 0.0, p, i, arcs.at(i));  // start length is overwritten later.
       found_first_start = true;
       DEBUG_PRINT("grad change detected. start at i = %lu", i);
     } else {
-      utils::avoidance::setEndData(al, shift, p, i, arcs.at(i));
+      setEndData(al, shift, p, i, arcs.at(i));
       al.id = getOriginalShiftLineUniqueId();
       merged_avoid_lines.push_back(al);
-      utils::avoidance::setStartData(
-        al, 0.0, p, i, arcs.at(i));  // start length is overwritten later.
+      setStartData(al, 0.0, p, i, arcs.at(i));  // start length is overwritten later.
       DEBUG_PRINT("end and start point found at i = %lu", i);
     }
   }
+
+  if (std::abs(backward_grad(N - 1)) > CREATE_SHIFT_GRAD_THR) {
+    const auto & p = path.points.at(N - 1).point.pose;
+    const auto shift = sl.shift_line.at(N - 1);
+    setEndData(al, shift, p, N - 1, arcs.at(N - 1));
+    al.id = getOriginalShiftLineUniqueId();
+    merged_avoid_lines.push_back(al);
+  }
+
   return merged_avoid_lines;
 }
 
+AvoidLineArray AvoidanceModule::fillShiftLineGap(const AvoidLineArray & input) const
+{
+  if (input.empty()) {
+    return input;
+  }
+
+  AvoidLineArray sorted_input = input;
+
+  alignShiftLinesOrder(sorted_input, false);
+
+  AvoidLineArray output = sorted_input;
+
+  {
+    const auto front = sorted_input.front();
+    const auto has_gap = front.start_idx > avoidance_data_.ego_closest_path_index;
+
+    if (has_gap) {
+      AvoidLine new_line{};
+      new_line.start_shift_length = helper_.getEgoLinearShift();
+      new_line.start_longitudinal = 0.0;
+      new_line.end_shift_length = front.start_shift_length;
+      new_line.end_longitudinal = front.start_longitudinal;
+      new_line.id = getOriginalShiftLineUniqueId();
+
+      output.push_back(new_line);
+    }
+  }
+
```

Comment:
how about using lambda function?
```suggestion
    auto createNewAvoidLine = [this](double start_shift_length, double start_longitudinal,
                                    double end_shift_length, double end_longitudinal) {
      AvoidLine newLine{};
      newLine.start_shift_length = start_shift_length;
      newLine.start_longitudinal = start_longitudinal;
      newLine.end_shift_length = end_shift_length;
      newLine.end_longitudinal = end_longitudinal;
      newLine.id = getOriginalShiftLineUniqueId();
      return new_line;
    };

    if (has_gap) {
      auto new_line = createNewAvoidLine(helper_.getEgoLinearShift(), 0.0, firstLine.start_shift_length, 
                                      firstLine.start_longitudinal);
      output.push_back(new_line);
    }
  }

  for (size_t i = 0; i < sorted_input.size() - 1; ++i) {
    const auto front = sorted_input.at(i);
    const auto back = sorted_input.at(i + 1);
    const auto has_gap = back.start_longitudinal - front.end_longitudinal > 0.0;

    if (has_gap) {
      auto new_line = createNewAvoidLine(currentLine.end_shift_length, currentLine.end_longitudinal, 
                                        nextLine.start_shift_length, nextLine.start_longitudinal);
```

Reply:
Use lambda in https://github.com/autowarefoundation/autoware.universe/pull/3852/commits/96af6872d7bea07db5632efa38670c54252a683f.